### ISSUE TITLE
GH Action isn't failing on `hugo server` when there's a build error

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-docs#299]
In PR #298, a commit was made that correctly wouldn't build.  This is what Hugo did when run locally on the developer machine:

```bash
➜  cht-docs git:(remote-login-updates) ../hugo server 
Building sites … ERROR 2020/09/01 16:12:08 [see-also] REF_NOT_FOUND: Ref "apps/concepts/access/#remote-login": "apps/examples/training.md": page not found
ERROR 2020/09/01 16:12:08 [see-also] REF_NOT_FOUND: Ref "apps/examples/training/": "apps/concepts/access.md": page not found
ERROR 2020/09/01 16:12:09 [see-also] REF_NOT_FOUND: Ref "apps/concepts/access/#remote-login": "apps/reference/api.md": page not found
Built in 776 ms
Error: Error building site: logged 3 error(s)
```

This is correct!  However, when GH Actions ran CI, and hit the same error [when running `hugo server`](https://github.com/medic/cht-docs/blob/00f288aecf31f948bf5e9b280dbe0912ef8dd091/.github/workflows/ci.yml#L42), it did not fail as it should have at the  "All Branches - Serve Hugo site" step. This makes the next step " All Branches - Run Muffet link checker" have a false positive:

![server hugo NOT OK](https://user-images.githubusercontent.com/8253488/91915257-e4f46100-ec6e-11ea-9fae-22667dcf8979.png)

We should ensure we fail the build in GH Actions before running `muffet` so doc authors know how to fix their builds.

**Comment by mrjones-plip** [comment]
@n-orlowski  & @garethbowen  - bad builds will now break with in hopefully logical ways instead of [illogical ways like before](https://github.com/medic/cht-docs/pull/298#pullrequestreview-480135106)!  You can see an [example here](https://github.com/medic/cht-docs/runs/1074283936).

tl;dr - there's a new " All Branches - Test For Broken Builds in Hugo" section and it will error with things like `ERROR [see-also] REF_NOT_FOUND: Ref "apps/reference/app-settings/user-roles/": "apps/concepts/users.md": page not found` which I think is clear. Lemme know if we should [update docs](https://docs.communityhealthtoolkit.org/contribute/docs/workflow/) or do any other outreach!

FYI @abbyad .
